### PR TITLE
Shipping address checkout page missing Set as default checkbox (GH-7153)

### LINKED ISSUE #7153: Shipping address checkout page missing Set as default checkbox
**Environment Details**
Spartacus: develop branch
Browser: Chrome
Device: mac
Happens on: http://dev-com-10.accdemo.b2c.ydev.hybris.com:4200/electronics-spa/en/USD
Calydon: **disabled**

**Steps to Reproduce**
- Have a saved shipping address
- Add item to cart.
- Proceed to checkout
- On shipping address checkout page, select "Add new address" button.

**Actual Result**
There's no "Set as default" checkbox. 
we only get it for the first address saved.
But we can change the default address through address book.
but was removing the checkbox intended?

Screen Shot 2020-04-02 at 11 11 44 AM

### PR DESCRIPTION
**Pull request by qjimmy**
Closes GH-7153

Retested manually both with the case of no current address and the case where the user already has an address. Also retested manually for guest checkout, checkbox is not there as intended. Found out my last previous change made no sense because updating this branch overwrote my code

- change was made on line 54

<hr />

should have no checkbox with guest checkout:
Screen Shot 2020-04-08 at 4 45 26 PM

<hr />

should have checkbox if not guest checkout:
Screen Shot 2020-04-08 at 4 54 14 PM

### CODE REVIEW

**Comment by bgambocjaviniar** [comment]
I think you need to fix your prettier first

you can do so, with the following

upgrade manually, and do the following steps Stan provided

or 

go to develop, git pull, delete node modules, and install the new versions from the updated package.json. (fastest is to just clone a new one and install a fresh one :')) haha

**Review by LTiger14 — CHANGES_REQUESTED** [review]
I can see the “Set as default” checkbox is only visible if you don’t already have a saved address. I you click add an address there is no checkbox. From the ticket it seems there should be a checkbox even for the “Add new address”.

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts [hunk 1]
```diff
@@ -81,24 +80,26 @@ export class AddressFormComponent implements OnInit, OnDestroy {
...
+      isocode: [null, Validators.required],
+    }),
     postalCode: ['', Validators.required],
     phone: '',
-    defaultAddress: [false],
   });
 
   constructor(
-    protected fb: FormBuilder,
+    private fb: FormBuilder,
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: These should be protected

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts [hunk 2]
```diff
@@ -195,29 +198,24 @@ export class AddressFormComponent implements OnInit, OnDestroy {
...
-        this.regionsSub = this.regions$.pipe(take(1)).subscribe((regions) => {
-          const obj = regions.find(
-            (region) =>
-              region.isocode ===
-              this.addressForm.controls['region'].value.isocode
-          );
-          Object.assign(this.addressForm.value.region, {
-            isocodeShort: obj.isocodeShort,
-          });
+    if (this.addressForm.controls['region'].value.isocode) {
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: Why did you remove the check for `this.addressForm.valid`?

Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: Why did you change `this.addressForm.get('region').value.isocode` to `this.addressForm.controls['region'].value.isocode`? The get seems cleaner

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts [hunk 3]
```diff
@@ -69,9 +70,7 @@ export class AddressFormComponent implements OnInit, OnDestroy {
   suggestedAddressModalRef: ModalRef;
 
   addressForm: FormGroup = this.fb.group({
-    country: this.fb.group({
-      isocode: [null, Validators.required],
-    }),
+    defaultAddress: [false],
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: This change is not required and should be left out to keep a cleaner diff.

File: projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts [hunk 4]
```diff
@@ -81,24 +80,26 @@ export class AddressFormComponent implements OnInit, OnDestroy {
     region: this.fb.group({
       isocode: [null, Validators.required],
     }),
+    country: this.fb.group({
```
Comment (LTiger14) [inline projects/storefrontlib/src/cms-components/checkout/components/shipping-address/address-form/address-form.component.ts]: This is also not required.

**Review by LTiger14 — CHANGES_REQUESTED** [review]
Upon submit the request to add a new address I get this console error.
Screen Shot 2020-05-04 at 3 26 40 PM
 Not just in your branch but also develop. This is not part of this issue. See if it's a quick thing that could be included in this PR. If not please open an issue.

**Review by LTiger14 — APPROVED** [review]

**Pull request merged**